src/app/core/duplicate-records/duplicate-records.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { EntityMapperService } from '../entity/entity-mapper/entity-mapper.service'; 
+import { EntityRegistry } from '../entity/database-entity.decorator';
+import { EntitySchemaService } from '../entity/schema/entity-schema.service';
+import { Entity } from '../entity/model/entity';
+import { MatSnackBar, MatSnackBarHorizontalPosition, MatSnackBarVerticalPosition } from '@angular/material/snack-bar';
+import { ConfigService } from '../config/config.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class DuplicateRecordService {
+  get: jasmine.Spy<jasmine.Func>;
+  constructor(
+    private entitymapperservice: EntityMapperService,
+    private entityTypes: EntityRegistry,
+    private entityService: EntitySchemaService,
+    private snackBar: MatSnackBar,
+    private config: ConfigService
+  ) {}
+
+  showFlashMessage(message: string) {
+    const horizontalPosition: MatSnackBarHorizontalPosition = 'center';
+    const verticalPosition: MatSnackBarVerticalPosition = 'top';
+    this.snackBar.open(message, 'Close', {
+      duration: 3000, 
+      horizontalPosition: horizontalPosition,
+      verticalPosition: verticalPosition,
+    });
+  }
+
+  async duplicateRecord(sourceData: any, schemaName: string) {
+    if (!sourceData || sourceData.length === 0) {
+      const flashMessage = this.config.getConfig("flashMessage") as { message: string };
+      this.showFlashMessage(flashMessage.message);
```

Comment:
thanks for trying to make this configurable!

However, I think this message can be consistent across our clients and should always say the same thing. Translations are handled using `$localize` (see https://aam-digital.github.io/ndb-core/documentation/additional-documentation/how-to-guides/build-localizable-(translatable)-ui.html), so let's have the message hard-coded here directly instead of using config.